docker/git-puller/bin/entrypoint.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+set -e
+
+if [[ $# -lt 1 ]]; then
+  echo "Expected 1 or more parameters but got $#."
+  cat /README.md
+  exit 1
+fi
+
+GIT_REPOS=$@
+
+# Tells git to use the provided private key and automatically (without prompt)
+# add the new hosts keys to the known hosts file
+export GIT_SSH_COMMAND="ssh -o StrictHostKeyChecking=no -i ${PRIVATE_KEY_PATH}"
+
+# Provides private key passphrase instead of prompting the user
+export SSH_ASKPASS=/ssh-askpass.sh
+
+# Needs setting DISPLAY so that the script specified by SSH_ASKPASS is run
+export DISPLAY=:
+
+cd /git-puller
+
+for repo in $GIT_REPOS; do
+  if [ -z "$GIT_CLONE_BARE" ]; then
+    git clone ${repo}
+  else
+    git clone --bare ${repo}
+  fi
+done
```

Comment:
you could use variable substitution for a shorter version
```bash
BARE_ARG=${GIT_CLONE_BARE:+"--bare"}
for repo in $GIT_REPOS; do
    git clone ${BARE_ARG} ${repo}
done
````

Reply:
That's really nice 👍 .